File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    // n이 되는 연속된 자연수 조합 
    let count = 0;
    
    const comb = []
    
    let num = 1;
    
    let k = n
    
    while(k > 0){
        k  = (n - (num * (num - 1) / 2)) / num;
        
        if(k < 1){
            break;
        }
        
        console.log("값", k, Number.isInteger(k));
        
        if(Number.isInteger(k)){
            count++;    
        }
        
        num ++;
    }
    
    return count;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    count = 1
    
    i = 1
    prefix = 0
    while True:
        prefix += i
        if (n - prefix) / (i+1) < 1:
            break
        if (n-prefix) % (i+1) == 0:
            count += 1
        
        i += 1
    
    
    return count